Failure to compile GMT on Apple Silicon

<!--To help us understand and resolve your issue, please fill out the form to the best of your ability. Feel free to delete the sections that do not apply.-->

**Description of the problem**

I am testing our build process on a Mac mini with an M1 chip (ARM architecture).  I am doing this without GDAL for now since at least for MacPorts the geos library fails to build (and libpvx does not support arm64 so **ffmpeg** cannot be installed - these have tickets and will be fixed soon I hope).

I anticipated this compilation Eror:

```
../src/gmt_common_sighandler.c:178:14: error: no member named '__eip' in 'struct __darwin_arm_thread_state64'
        array [0] = UC_IP (uc); /* caller's address */
                    ^~~~~~~~~~
../src/gmt_common_sighandler.c:62:56: note: expanded from macro 'UC_IP'
#   define UC_IP(uc) ((void *) (uc)->uc_mcontext->__ss.__eip)
                               ~~~~~~~~~~~~~~~~~~~~~~~ ^
```

Basically, I need to the `__arm64__` equivalent of

`#   define UC_IP(uc) ((void *) (uc)->uc_mcontext->__ss.__rip)`


for the M1 chip.  Could anyone with knowledge point me to a solution?

--- a/src/gmt_common_sighandler.c
+++ b/src/gmt_common_sighandler.c
@@ -58,6 +58,8 @@ void backtrace_symbols_fd(void *const *buffer, int size, int fd) {
 # if __DARWIN_UNIX03
 #  ifdef __x86_64__
 #   define UC_IP(uc) ((void *) (uc)->uc_mcontext->__ss.__rip)
+#  elif __arm64__	/* Apple Silicon, e.g. M1 */
+#   define UC_IP(uc) ((void *) (uc)->uc_mcontext->__ss.__pc)
 #  else
 #   define UC_IP(uc) ((void *) (uc)->uc_mcontext->__ss.__eip)
 #  endif

